modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -321,13 +320,22 @@ private AsyncCursor<List<Object>> executeExplain(ExplainPlan plan) {
     private void onMessage(String nodeName, QueryStartRequest msg) {
         assert nodeName != null && msg != null;
 
-        DistributedQueryManager queryManager = queryManagerMap.computeIfAbsent(msg.queryId(), key -> {
-            BaseQueryContext ctx = createQueryContext(key, msg.schema(), msg.parameters());
+        CompletableFuture<?> fut = sqlSchemaManager.waitActualSchema(msg.lastVersion());
 
-            return new DistributedQueryManager(ctx);
-        });
+        fut.whenComplete((mgr, e) -> {
+            if (e != null) {
+                LOG.error("Schema operation error", e);
```

Comment:
if we failed to acquire schema of required version, why should be proceed execution of a query?

Reply:
ok let`s throw exception here.

Reply:
I think there should be a guarantee that QueryStartResponse will be sent back in case of error.